hoomd/SnapshotSystemData.cc

```diff
@@ -39,6 +39,27 @@ void SnapshotSystemData<Real>::replicate(unsigned int nx, unsigned int ny, unsig
     pair_data.replicate(n, old_n);
     }
 
+template<class Real> void SnapshotSystemData<Real>::wrap()
+    {
+    for (unsigned int i = 0; i < particle_data.size; i++)
+        {
+        auto const frac = global_box.makeFraction(particle_data.pos[i]);
+        auto modulus_positive = [](auto x){ return std::fmod(std::fmod(x, Real(1.0)) + Real(1.0), Real(1.0)); };
+        auto const wrapped = vec3<Real>(
+            modulus_positive(frac.x),
+            modulus_positive(frac.y),
+            modulus_positive(frac.z)
+            );
+        particle_data.pos[i] = global_box.makeCoordinates(wrapped);
+        auto const img = make_int3(
+            std::floor(frac.x),
+            std::floor(frac.y),
+            std::floor(frac.z)
```

Comment:
```suggestion
            static_cast<int>(std::floor(frac.x)),
            static_cast<int>(std::floor(frac.y)),
            static_cast<int>(std::floor(frac.z))
```
HOOMD should compile with no warnings, so you need the explicit cast here to tell the compiler it is OK to convert a Real to an int. Also, I'd suggest a comment explaining why floor is the right operation here. It's not intuitive why a simple static cast to an int doesn't work.